### ISSUE TITLE
0-Confidence bug

### ISSUE THREAD

**Issue by blumx116** [blumx116/VotingIsAllYouNeed#73]
One Person Predicts (confidence = 0)
Other person predicts (confidence = non-zero)
Only those two people predict

The person who predicts with non-zero confidence will lose money with the 95 rule (or it gives division by 0), whereas they should just get their money back.

This could be solved by having an agent return `None` as it's `Weighted` when it places its bet if the confidence is 0, then filtering out all of the `Weighted`s where the confidence is zero

**Comment by alnah005** [comment]
This problem could be there under the 100% rule couldn't it? If yes, then we should probably make sure its fixed there as well.

**Comment by blumx116** [comment]
Yeah, I think that the problem also occurs in the 95% rule, so we should definitely check that (and probably make tests for it as well). I think that the above scheme with `Weighted` should fix both at once.
